runtime/service/src/main/java/org/apache/polaris/service/quarkus/config/QuarkusProducers.java

```diff
@@ -170,6 +175,34 @@ public UserSecretsManagerFactory userSecretsManagerFactory(
     return userSecretsManagerFactories.select(Identifier.Literal.of(config.type())).get();
   }
 
+  @Produces
+  @Singleton
+  @Identifier("http-client-s3")
+  public SdkHttpClient sdkHttpClient(S3AccessConfig config) {
+    ApacheHttpClient.Builder httpClient = ApacheHttpClient.builder();
```

Comment:
why only ApacheHttpClient and not URLConnection client ?

Reply:
I can make it configurable in runtime if you prefer. Let's do that as a follow-up PR, though.

Reply:
`ApacheHttpClient` is more capable in general, AFAIK.

Reply:
Iceberg supports both, hence coming from there, Its good to have alternative specially considering how they behave. for example The Apache HTTP connection will always read the [entire stream](https://github.com/apache/httpcomponents-core/blob/2135596b1319910d56efb148566ecbbec0884638/httpcore/src/main/java/org/apache/http/impl/io/ContentLengthInputStream.java#L108) on close, which results in much more data being read than needed. The URL connection client does not behave this way.  for more details : https://github.com/apache/iceberg/pull/7262

Reply:
Certainly. If do not mind supporting both, but let's do that in a follow-up PR to avoid bloating the scope of this PR (the main idea in this PR is not the ApacheHttpClient client, but supporting multiple endpoints).